checkov/kustomize/runner.py

```diff
@@ -118,22 +129,89 @@ def mutate_kubernetes_results(
                 # Fix file path to repo relative path
                 if self.original_root_dir:
                     repo_dir = str(pathlib.Path(self.original_root_dir).resolve())
+
                     if realKustomizeEnvMetadata['filePath'].startswith(repo_dir):
                         file_path = realKustomizeEnvMetadata['filePath'][len(repo_dir):]
 
+            if self.checkov_allow_kustomize_file_edits:
+                caller_file_line_range, caller_file_path = self._get_caller_file_info(entity_context, k8_file,
+                                                                                      k8_file_path, resource_id,
+                                                                                      root_folder)
             code_lines = entity_context.get("code_lines")
             file_line_range = self.line_range(code_lines)
+
             record = Record(
                 check_id=check.id, bc_check_id=check.bc_id, check_name=check.name,
-                check_result=check_result, code_block=code_lines, file_path=file_path,
-                file_line_range=file_line_range,
+                check_result=check_result, code_block=code_lines,
+                file_path=file_path, file_line_range=file_line_range,
+                caller_file_path=caller_file_path, caller_file_line_range=caller_file_line_range,
                 resource=kustomizeResourceID, evaluations=variable_evaluations,
                 check_class=check.__class__.__module__, file_abs_path=absolute_file_path, severity=check.severity)
             record.set_guideline(check.guideline)
             report.add_record(record=record)
 
         return report
 
+    def _get_caller_file_info(self, entity_context: _EntityContext, k8_file: str, k8_file_path: str, resource_id: str,
+                              root_folder: str | None) -> tuple[tuple[int, int] | None, str | None]:
+        origin_relative_path = entity_context['origin_relative_path']
```

Comment:
```suggestion
        origin_relative_path = entity_context.get('origin_relative_path')
```

Reply:
actually entityContext is a defined type so it's not needed :)
you can look for `_EntityContext`